test/middlewares/build-include.spec.ts

```diff
@@ -0,0 +1,196 @@
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const db = require("./../../example/db");
+
+describe("Build Include Middleware", () => {
+  let req: Partial<SequelizeQueryParserRequestInterface> & {
+    sequelizeQueryParser: any;
+    query: any;
+  };
+  let res: Partial<Response>;
+  let next: jest.Mock;
+
+  beforeEach(() => {
+    req = {
+      sequelizeQueryParser: {
+        model: db["Municipality"],
+      },
+      query: {}, // Define query here
+    };
+    res = {
+      status: jest.fn().mockReturnThis(), // Ensure 'status' returns 'this' (res) for chaining
+      json: jest.fn(),
+    } as unknown as Response; // Explicitly define res as Response
+    next = jest.fn();
+  });
+
+  it("should yield a valid include object", () => {
+    req.query.include = "Province";
+    const controlValue = [
+      {
+        association: "Province",
+        required: false,
+      },
+    ];
+    buildInclude(
+      req as SequelizeQueryParserRequestInterface,
+      res as Response,
+      next
+    );
+    expect(req.sequelizeQueryParser.include).toBeDefined();
+    expect(req.sequelizeQueryParser.include).toStrictEqual(controlValue);
+    expect(next).toHaveBeenCalled();
+  });
+
+  it("should yield an empty include object by default", () => {
+    const controlValue = [];
+    buildInclude(
+      req as SequelizeQueryParserRequestInterface,
+      res as Response,
+      next
+    );
+    expect(req.sequelizeQueryParser.include).toBeDefined();
+    expect(req.sequelizeQueryParser.include).toStrictEqual(controlValue);
+    expect(next).toHaveBeenCalled();
+  });
+
+  it("should yield an empty include object when req.query.include is an empty string", () => {
+    req.query.include = "";
+    const controlValue = [];
+    buildInclude(
+      req as SequelizeQueryParserRequestInterface,
+      res as Response,
+      next
+    );
+    expect(req.sequelizeQueryParser.include).toBeDefined();
+    expect(req.sequelizeQueryParser.include).toStrictEqual(controlValue);
+    expect(next).toHaveBeenCalled();
+  });
+
+  it("should yield a valid include object with deep associations", () => {
+    req.sequelizeQueryParser.model = db["SocialEvent"];
+    req.query.include = "Document,Album.Images";
+    const controlValue = [
+      {
+        association: "Document",
+        required: false,
+      },
+      {
+        association: "Album",
+        include: [
+          {
+            association: "Images",
+            required: false,
+          },
+        ],
+        required: false,
+      },
+    ];
+    buildInclude(
+      req as SequelizeQueryParserRequestInterface,
+      res as Response,
+      next
+    );
+    expect(req.sequelizeQueryParser.include).toBeDefined();
+    expect(req.sequelizeQueryParser.include).toStrictEqual(controlValue);
+    expect(next).toHaveBeenCalled();
+  });
+
+  it("should yield a valid include object with deep associations and sibling associations", () => {
```

Comment:
Fix you code to pass this test case. The resulting include must have 2 associatons inside Events include. Your associations array has a one row per leaf associations (association without children), even if those associations belongs to the same parent